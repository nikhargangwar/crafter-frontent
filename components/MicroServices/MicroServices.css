@import url('https://fonts.googleapis.com/css2?family=Big+Shoulders+Inline+Display:wght@500;600;700&family=Big+Shoulders+Text:wght@700&family=Sofia+Sans:wght@700&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Bebas+Neue&display=swap');

.microservices {
  background-color: #211E1E !important;
  color: white !important;
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 1px solid black;
  height: 100%;
  width: 18vw;
  min-width: 280px;
  font-family: 'Big Shoulders Inline Display', cursive !important;
  font-family: 'Big Shoulders Text', cursive !important;
  font-family: 'Sofia Sans', sans-serif !important;
  overflow-x: scroll;
}

.microservices::-webkit-scrollbar {
  display: none;
}

.microservices-header {
  height: 6vh;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 1.8rem !important;
  border-bottom: 2px solid #18CCCC;
  font-family: 'Bebas Neue', cursive;
  text-align: center;
  padding: 10px 0;
  width: 100%;
  margin: 3px;
}

.heading {
  font-size: medium;
  display: flex;
  flex-direction: column;
}

hr {
  width: 100%;
  border: solid black;
}

.drop-down-service {
  width: 100%;
}

.microservice-drag-drop {
  display: flex;
  flex-direction: column;
  align-items: center;
  /* justify-content:space-evenly ; */
  width: 100%;
  margin: 2vh 0;
  font-weight: bold;
  font-size: 1.5rem;
}

.service {
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-direction: row;
  height: 8vh !important;
  width: 70%;
  margin: 1vh 0 0 0 !important;
  color: rgb(255, 255, 255) !important;
  border-radius: 10px;
  font-weight: bold;
  font-size: 1em;
  background-color: none;
  font-family: 'Bebas Neue', cursive;
  border: 2px solid #18CCCC;
  padding: 0 20px;
}


.service:hover {
  color: rgb(255, 255, 255) !important;
  cursor: pointer;
  border: 2px solid black !important;
  background-color: #464646;
}

/* @media screen and (max-width: 1600px) {
  .microservices {
    width: 20% !important;
  }
  .microservices-header {
    font-size: 2.5vw !important;
  }
  .service {
    font-size: 1.5vw !important;
  }

} */