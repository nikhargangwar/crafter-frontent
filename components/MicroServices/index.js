import React, { useEffect, useState } from 'react';
import { makeRequest } from '../../utils';
import { apiEndPoints } from '../../constants';
import { useNavigate } from 'react-router-dom';
import { capitalizeEachWord } from '../../utils/ServiceHandler';
import { Tooltip } from '@mui/material';

const MicroServices = () => {
  const [microservices, setMicroservices] = useState([]);
  const navigate = useNavigate();

  useEffect(() => {
    const token = localStorage.getItem('jwt_token');
    makeRequest.makeRequest(apiEndPoints.GET_SYSTEM_COMPONENTS(), navigate, token)
      .then((response) => {
        setMicroservices(response);
      });
  }, []);

  const onDragStart = (event,nodeType,pluginId) => {
    event.dataTransfer.setData('application/reactflow', nodeType);
    event.dataTransfer.setData('pluginId', pluginId);
    event.dataTransfer.effectAllowed = 'move';
  };

  return (
    <div className='flex flex-col items-center h-[70vh] min-w-[200px] rounded-3xl  bg-[#211E1E] ml-10 my-10 border-[3px] border-solid border-[#18CCCC] overflow-y-scroll'>
      <div className='flex items-center justify-center h-[6vh] text-[23px] text-center w-full py-2 px-0 m-1 text-white border-solid border-b-2 border-b-[#18CCCC] font-["Bebas_Neue"]'>SYSTEM COMPONENTS</div>
      {microservices.map((microservice) => (
        <Tooltip title={microservice.metadata.description} placement='right' key={microservice.id}>
          <div className='flex flex-col items-center w-[120px] h-[150px] m-4 border-2 border-[#18CCCC] rounded-xl mx-0 font-bold text-2xl hover:text-white hover:cursor-pointer hover:bg-[#464646] hover:border-black'>
            <div draggable onDragStart={(event) => onDragStart(event, 'straight',microservice.id)} 
              className='flex flex-col items-center justify-between h-[vh] w-[70%] mt-[1vh] text-white  font-bold text-base bg-transparent px-2 py-0 '>
              <img className='w-12' src={`${process.env.REACT_APP_API_GATEWAY_URL}/minio/images/${microservice.metadata.icon_node_filename}`}/>
              <p className='text-center text-[15px]'>{capitalizeEachWord(microservice.metadata.name.replace(/-/g,' '))}</p>
            </div>
          </div>
        </Tooltip>
      ))} 
    </div>
  );
};

export default MicroServices;