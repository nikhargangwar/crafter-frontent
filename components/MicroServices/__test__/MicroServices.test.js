import React from 'react';
import { render } from '@testing-library/react';
import MicroServices from '../index';
// import {useNavigate} from 'react-router-dom';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));
describe('MicroServices', () => {
  it('should render correctly', () => {
    const { container } = render(<MicroServices />);
    expect(container).toMatchSnapshot();
  });
});