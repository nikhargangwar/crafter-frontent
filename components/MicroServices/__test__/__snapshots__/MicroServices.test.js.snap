// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MicroServices should render correctly 1`] = `
<div>
  <div
    class="flex flex-col items-center h-[70vh] min-w-[200px] rounded-3xl  bg-[#211E1E] ml-10 my-10 border-[3px] border-solid border-[#18CCCC] overflow-y-scroll"
  >
    <div
      class="flex items-center justify-center h-[6vh] text-[23px] text-center w-full py-2 px-0 m-1 text-white border-solid border-b-2 border-b-[#18CCCC] font-[\\"Bebas_Neue\\"]"
    >
      SYSTEM COMPONENTS
    </div>
  </div>
</div>
`;
